src/ui/components/ExperimentBrowserPage.tsx

```diff
@@ -179,84 +185,67 @@ const ExperimentBrowserPage: FC = () => {
           </tr>
         </thead>
         <tbody>
-          {experiments.map((experiment) => (
-            <tr key={experiment.id}>
-              <td className="align-middle ps-0 py-3 w-50">
-                <strong>{experiment.userFacingName}</strong>:{" "}
-                {experiment.userFacingDescription}
-              </td>
-              <td className="text-center align-middle px-2">
-                {experiment.channel}
-              </td>
-              <td className="text-center align-middle px-2">
-                {experiment.schemaVersion}
-              </td>
-              <td className="text-center align-middle px-2">
-                {experiment.isEnrollmentPaused
-                  ? "Enrolling"
-                  : "Enrollment Paused"}
-              </td>
-              <td className="text-end align-middle wide-column">
-                <Container className="d-flex align-items-center">
-                  <Form.Select
-                    value={selectedBranches[experiment.id]}
-                    onChange={(e) =>
-                      handleBranchChange(experiment.id, e.target.value)
-                    }
-                    className="grey-border small-font rounded p-2 m-0 font-monospace"
-                  >
-                    <option value="">Select branch</option>
-                    {experiment.branches?.map((branch) => (
-                      <option key={branch.slug} value={branch.slug}>
-                        {branch.slug}
-                      </option>
-                    ))}
-                  </Form.Select>
-                  {status === "Live" ? (
+          {experiments.map((experiment) => {
+            const onEnroll = () =>
+              handleEnroll(experiment.id, selectedBranches[experiment.id]);
+            const onGenerateTestId = () =>
+              handleGenerateTestIds(
+                experiment.id,
+                selectedBranches[experiment.id],
+              );
+
+            return (
+              <tr key={experiment.id}>
+                <td className="align-middle ps-0 py-3 w-50">
+                  <strong>{experiment.userFacingName}</strong>:{" "}
+                  {experiment.userFacingDescription}
+                </td>
+                <td className="text-center align-middle px-2">
+                  {experiment.channel}
+                </td>
+                <td className="text-center align-middle px-2">
+                  {experiment.schemaVersion}
+                </td>
+                <td className="text-center align-middle px-2">
+                  {experiment.isEnrollmentPaused
+                    ? "Enrolling"
+                    : "Enrollment Paused"}
+                </td>
+                <td className="text-end align-middle wide-column">
+                  <Container className="d-flex align-items-center">
+                    <Form.Select
+                      value={selectedBranches[experiment.id]}
```

Comment:
id and slug are the same, but slug is canonical as far as Nimbus and Experimenter are concerned.